gitbook-docs/urls_etc.md

```diff
@@ -69,14 +68,28 @@ If a server does not support some streaming options listed in here it must respo
 
 See [Subscription Protocol](subscription_protocol.html) for more details.
 
-##### Connection Hello
+#### Streaming Connection "Hello"
 
-Upon connection a 'hello' message is sent as follows:
+Upon accepting a connection a server MUST send a 'hello' message before any other communicaiton is sent. The format of the 'hello' message is as follows:
 
 ```json
 {
   "version": "1.1.2",
+  "source": "actisense.ngw-1.1234567",
   "timestamp": "2015-04-13T01:13:50.524Z",
   "self": "123456789"
 }
 ```
+
+"version" - The 'hello' message MUST contain a "version" element which MUST have a string value identifying the lowest version of signalk-delta that defines the format of the messages that will be sent by the server.
+
+"source" - The 'hello' message MUST contain a "source" element which MUST have a string value that identifies the server as follows: manufacturer/publisher.device/application.uuid, where:
```

Comment:
To be a blt clearer, the 'hello' message comes from whatever is initiating the SK stream. This could be a client sending to a server, or a server responding to a client. In either case, the sender wants to identify themselves as the source of the information which is what the type in /definitions is for.

For example, a battery monitoring device discovers a SK server on its network, connects, sends a 'hello' identifying itself and then starts sending readings. It doesn't have a RTC so it sends a base timestamp of '0' and then time deltas in ms with the 'update' messages. The SK server includes information from the source value in its model and in delta messages it sends out.

A client discovers the SK server, connects to it and subscribes to a stream. The SK server identifies itself in the 'hello' and then forwards updates as it receives messages over SK, N2K or '183. The client uses the source information from the battery monitor to distinguish updates for the house bank from those from the engine bank.

A intermediate SK server can act as a forwarding proxy for SK delta without needing be configured with what every device it.